editor.planx.uk/src/ui/shared/CharacterCounter.tsx

```diff
@@ -0,0 +1,76 @@
+import Typography from "@mui/material/Typography";
+import { debounce } from "lodash";
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { FONT_WEIGHT_SEMI_BOLD } from "theme";
+
+export type Props = {
+  characterLimit: number;
+  characterCount: number;
+  error: boolean;
+};
+
+export const CharacterCounter: React.FC<Props> = (props) => {
+  const [characterLimitAnnouncement, setCharacterLimitAnnouncement] =
+    useState<number>(0);
+
+  const hintTextBool = useRef(true);
+  const updateCharacterCount = useCallback(
```

Comment:
Not super familiar with best practices on `useCallback` and `debounce` but I wanted it to only be called once on mounting in React and then called after a certain time period of typing to reduce noise in the screen reader.

Keen for alternates / thoughts